src/query/parser/promql/matchers.go

```diff
@@ -90,27 +90,31 @@ func NewAggregationOperator(expr *promql.AggregateExpr) (parser.Params, error) {
 		Without:      expr.Without,
 	}
 
-	op := getAggOpType(opType)
-	if op == common.UnknownOpType {
+	switch op := getAggOpType(opType); op {
+	case common.UnknownOpType:
 		return nil, fmt.Errorf("operator not supported: %s", opType)
-	}
-
-	if op == aggregation.BottomKType || op == aggregation.TopKType {
+	case aggregation.BottomKType, aggregation.TopKType:
 		val, err := resolveScalarArgument(expr.Param)
 		if err != nil {
 			return nil, err
 		}
 
 		nodeInformation.Parameter = val
 		return aggregation.NewTakeOp(op, nodeInformation)
-	}
+	case aggregation.QuantileType:
```

Comment:
Mini nit: move this case below `CountValuesType`. It ends up with `NewAggregationOp`, just like the `default` case, so it will be a bit easier seeing them next to each other.